cardano_node_tests/tests/test_plutus_v2_delegation.py

```diff
@@ -0,0 +1,408 @@
+    """Deregister stake address."""
+    src_payment_balance = cluster_obj.get_address_balance(pool_user.payment.address)
+    reward_balance = cluster_obj.get_stake_addr_info(pool_user.stake.address).reward_account_balance
+
+    # create stake address deregistration cert
+    stake_addr_dereg_cert = cluster_obj.gen_stake_addr_deregistration_cert(
+        addr_name=f"{temp_template}_addr0",
+        stake_script_file=pool_user.stake.script_file,
+    )
+
+    # withdraw rewards to payment address, deregister stake address
+    withdrawal_script = clusterlib.ScriptWithdrawal(
+        txout=clusterlib.TxOut(address=pool_user.stake.address, amount=-1),
+        reference_txin=reference_script_utxos[0],
+        collaterals=[collaterals[0]],
+        redeemer_file=redeemer_file,
+    )
+    dereg_cert_script = clusterlib.ComplexCert(
+        certificate_file=stake_addr_dereg_cert,
+        reference_txin=reference_script_utxos[0],
+        collaterals=[collaterals[1]],
+        redeemer_file=redeemer_file,
+    )
+
+    tx_files = clusterlib.TxFiles(signing_key_files=[pool_user.payment.skey_file])
+
+    tx_raw_output = cluster_obj.build_tx(
+        src_address=pool_user.payment.address,
+        tx_name=f"{temp_template}_dereg_withdraw",
+        txins=txins,
+        tx_files=tx_files,
+        complex_certs=[dereg_cert_script],
+        fee_buffer=2_000_000,
+        script_withdrawals=[withdrawal_script],
+    )
+    # calculate cost of Plutus script
+    plutus_cost = cluster_obj.calculate_plutus_script_cost(
+        src_address=pool_user.payment.address,
+        tx_name=f"{temp_template}_dereg_withdraw",
+        txins=txins,
+        tx_files=tx_files,
+        complex_certs=[dereg_cert_script],
+        fee_buffer=2_000_000,
+        script_withdrawals=[withdrawal_script],
+    )
+    tx_signed = cluster_obj.sign_tx(
+        tx_body_file=tx_raw_output.out_file,
+        signing_key_files=tx_files.signing_key_files,
+        tx_name=f"{temp_template}_reg_deleg",
+    )
+
+    cluster_obj.submit_tx(tx_file=tx_signed, txins=tx_raw_output.txins)
+
+    # check that the key deposit was returned and rewards withdrawn
+    assert (
+        cluster_obj.get_address_balance(pool_user.payment.address)
+        == src_payment_balance
+        - tx_raw_output.fee
+        + reward_balance
+        + cluster_obj.get_address_deposit()
+    ), f"Incorrect balance for source address `{pool_user.payment.address}`"
+
+    # check that the stake address is no longer delegated
+    stake_addr_info = cluster_obj.get_stake_addr_info(pool_user.stake.address)
+    assert not stake_addr_info.delegation, f"Stake address is still delegated: {stake_addr_info}"
+
+    tx_db_dereg = dbsync_utils.check_tx(cluster_obj=cluster_obj, tx_raw_output=tx_raw_output)
+    if tx_db_dereg:
+        assert pool_user.stake.address in tx_db_dereg.stake_deregistration
+
+        # compare cost of Plutus script with data from db-sync
+        dbsync_utils.check_plutus_cost(
```

Comment:
I am getting different values on plutus cost:
```
raise AssertionError("\n".join(errors))
E           AssertionError: time: 218855869 vs 213184888
E           space: 686154 vs 670528
E           fixed cost: 55371 vs 54061
```

But don't understand why, already compare the transactions and doesn't look anything wrong:

tx_raw_output tx checked on dbsync:
```
['transaction', 'build', 
'--tx-in-collateral', '926cdbc4bb5f241723ed2ffd086b91671e8bc54343c72e20e97fb1f90ddde3fe#3', 
'--certificate-file', 'test_delegate_deregister_ci0_mku_addr0_stake_dereg.cert', 
'--certificate-tx-in-reference', '926cdbc4bb5f241723ed2ffd086b91671e8bc54343c72e20e97fb1f90ddde3fe#6', 
'--certificate-plutus-script-v2', 
'--certificate-reference-tx-in-redeemer-file', 'plutus/42.redeemer', 
'--tx-in-collateral', '926cdbc4bb5f241723ed2ffd086b91671e8bc54343c72e20e97fb1f90ddde3fe#2', 
'--withdrawal', 'stake_test17zwga8d8lq0re2gysheja0hunqfhezkzvzs89gqvf2h3gtgtc7vzc+11375945', 
'--withdrawal-tx-in-reference', '926cdbc4bb5f241723ed2ffd086b91671e8bc54343c72e20e97fb1f90ddde3fe#6', 
'--withdrawal-plutus-script-v2', 
'--withdrawal-reference-tx-in-redeemer-file', 'plutus/42.redeemer', 
'--protocol-params-file', 'pparams-qsgt.json', 
'--tx-in', '926cdbc4bb5f241723ed2ffd086b91671e8bc54343c72e20e97fb1f90ddde3fe#5', 
'--change-address', 'addr_test1yq89239mp3yt9ese9mkchvck66lvqersgfafxxdh80nhk95u36w607q78j5sfp0n96l0exqn0j9vyc9qw2sqcj40zsksdhp0g9', 
'--out-file', 'test_delegate_deregister_ci0_mku_dereg_withdraw_tx.body', 
'--babbage-era', '--testnet-magic', '42']
```

tx that calculates plutus cost:
```
['transaction', 'build', 
'--tx-in-collateral', '926cdbc4bb5f241723ed2ffd086b91671e8bc54343c72e20e97fb1f90ddde3fe#3', 
'--certificate-file', 'test_delegate_deregister_ci0_mku_addr0_stake_dereg.cert', 
'--certificate-tx-in-reference', '926cdbc4bb5f241723ed2ffd086b91671e8bc54343c72e20e97fb1f90ddde3fe#6', 
'--certificate-plutus-script-v2', 
'--certificate-reference-tx-in-redeemer-file', '/plutus/42.redeemer', 
'--tx-in-collateral', '926cdbc4bb5f241723ed2ffd086b91671e8bc54343c72e20e97fb1f90ddde3fe#2', 
'--withdrawal', 'stake_test17zwga8d8lq0re2gysheja0hunqfhezkzvzs89gqvf2h3gtgtc7vzc+11375945', 
'--withdrawal-tx-in-reference', '926cdbc4bb5f241723ed2ffd086b91671e8bc54343c72e20e97fb1f90ddde3fe#6', 
'--withdrawal-plutus-script-v2', 
'--withdrawal-reference-tx-in-redeemer-file', 'plutus/42.redeemer', 
'--protocol-params-file', 'pparams-qsgt.json', 
'--tx-in', '926cdbc4bb5f241723ed2ffd086b91671e8bc54343c72e20e97fb1f90ddde3fe#5', 
'--change-address', 'addr_test1yq89239mp3yt9ese9mkchvck66lvqersgfafxxdh80nhk95u36w607q78j5sfp0n96l0exqn0j9vyc9qw2sqcj40zsksdhp0g9', 
'--calculate-plutus-script-cost', 'test_delegate_deregister_ci0_mku_dereg_withdraw_plutus.cost', 
'--babbage-era', '--testnet-magic', '42']
```